discovery_engine/test/integration/search_test.dart

```diff
@@ -0,0 +1,240 @@
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import 'dart:io' show Directory;
+
+import 'package:test/test.dart';
+import 'package:xayn_discovery_engine/discovery_engine.dart'
+    show
+        ActiveSearchClosedFailed,
+        ActiveSearchClosedSucceeded,
+        ActiveSearchRequestFailed,
+        ActiveSearchRequestSucceeded,
+        ActiveSearchTermRequestFailed,
+        ActiveSearchTermRequestSucceeded,
+        DiscoveryEngine,
+        NextActiveSearchBatchRequestFailed,
+        NextActiveSearchBatchRequestSucceeded,
+        RestoreActiveSearchSucceeded,
+        SearchBy,
+        SearchFailureReason;
+
+import '../logging.dart' show setupLogging;
+import 'utils/helpers.dart'
+    show TestEngineData, expectEvent, initEngine, setupTestEngineData;
+import 'utils/local_newsapi_server.dart' show LocalNewsApiServer;
+
+void main() {
+  setupLogging();
+
+  const searchTerm = 'birds eat seeds';
+  const searchTerm2 = 'birds also eat insects';
+
+  group('News Search', () {
+    late LocalNewsApiServer server;
+    late TestEngineData data;
+    late DiscoveryEngine engine;
+
+    setUp(() async {
+      data = await setupTestEngineData();
+      server = await LocalNewsApiServer.start();
+      engine = await initEngine(data, server.port);
+    });
+
+    tearDown(() async {
+      await engine.dispose();
+      await server.close();
+      await Directory(data.applicationDirectoryPath).delete(recursive: true);
+    });
+
+    test('a new search can be opened', () async {
+      final result = expectEvent<ActiveSearchRequestSucceeded>(
+        await engine.requestQuerySearch(searchTerm),
+      );
+
+      expect(result.items, isNotEmpty);
+      expect(result.search.searchBy, SearchBy.query);
+      expect(result.search.searchTerm, searchTerm);
+
+      expect(
+        expectEvent<ActiveSearchTermRequestSucceeded>(
+          await engine.getActiveSearchTerm(),
+        ).searchTerm,
+        equals(searchTerm),
+      );
+    });
+
+    test('a new search can not be opened twice', () async {
+      expectEvent<ActiveSearchRequestSucceeded>(
+        await engine.requestQuerySearch(searchTerm),
+      );
+
+      expect(
+        expectEvent<ActiveSearchRequestFailed>(
+          await engine.requestQuerySearch(searchTerm2),
+        ).reason,
+        equals(SearchFailureReason.openActiveSearch),
+      );
+    });
+
+    test('we can request more results until we ran out of results', () async {
+      final result = expectEvent<ActiveSearchRequestSucceeded>(
+        await engine.requestQuerySearch(searchTerm),
+      );
+
+      expect(result.items, isNotEmpty);
+      expect(result.search.searchBy, SearchBy.query);
+      expect(result.search.searchTerm, searchTerm);
+
+      final nextResult = expectEvent<NextActiveSearchBatchRequestSucceeded>(
+        await engine.requestNextActiveSearchBatch(),
+      );
+
+      expect(nextResult.items, isNot(equals(result.items)));
```

Comment:
:art: is the `equals` (here and similarly in other places) necessary?

Reply:
it is not needed, as they did overload `isNot` :rocket: